clients/src/main/java/org/apache/kafka/server/authorizer/AclDeleteResult.java

```diff
@@ -81,10 +83,10 @@ public AclBinding aclBinding() {
         }
 
         /**
-         * Returns exception that resulted in failure to delete ACL binding.
+         * Returns any exception that resulted in failure to delete ACL binding.
          */
-        public ApiException exception() {
-            return exception;
+        public Optional<ApiException> exception() {
+            return exception == null ? Optional.empty() : Optional.of(exception);
         }
 
         /**
```

Comment:
Should we remove deleted() also?

Reply:
@omkreddy Thanks for the review. Makes sense, removed `deleted()` as well.